lib/src/zenohUTransport.cpp

```diff
@@ -185,12 +185,19 @@ UCode ZenohUTransport::sendQueryable(const UMessage &message) noexcept {
 
     auto uuidStr = UuidSerializer::serializeToString(message.attributes().reqid());
 
-    if (queryMap_.find(uuidStr) == queryMap_.end()) {
-        spdlog::error("failed to find UUID = {}", uuidStr);
-        return UCode::UNAVAILABLE;
-    }
+    z_owned_query_t query;
+    {
+        std::unique_lock<std::mutex> lock(queryMapMutex_);
 
-    auto query = queryMap_[uuidStr];
+        if (auto query_it = queryMap_.find(uuidStr);
+                query_it == queryMap_.end()) {
+            spdlog::error("failed to find UUID = {}", uuidStr);
+            return UCode::UNAVAILABLE;
```

Comment:
If you return but do not unlock, won't that be a problem?

Reply:
`std::unique_lock` is a scoped ownership wrapper for a mutex. It automatically unlocks when it goes out of scope.